eclipse-che/che-server
# fix: improve bitbucket OAuth flow to use more specific error messages

### What does this PR do?
Improve Bitbucket OAuth flow to: 

 * use more specific error messages (more specificly, if there are TLS keys misconfiguration);
 * allows to use simplified configuration (w/out OAuth1 setup) for public repositories;

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20335


### How to test this PR?
 - Setup Che and BitBucket on different clusters. BitBucket must use self-signed cert, and it is must not be imported into Che.
 - Perform app linking procedure as per doc;
 - Try to create workspace from an _public_ BitBucket repository.
 Expected result: workspace is created. Actual result: error is shown w/out any details.
 - Restart Che using test image `mshaposh/che-server:nightly`. Result after fix: Correct error message with explanation is returned.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
